sql/columndefault.go

```diff
@@ -80,7 +80,8 @@ func (e *ColumnDefaultValue) Eval(ctx *Context, r Row) (interface{}, error) {
 	}
 
 	if e.outType != nil {
-		if val, err = e.outType.Convert(val); err != nil {
+		var outOfRange bool
+		if val, outOfRange, err = e.outType.Convert(val); err != nil || outOfRange {
```

Comment:
Looks like we're swallowing the `err` error. I think we should throw `err` if it's not `nil`, and if `outOfRange` is `true`, return `ErrIncompatibleDefaultType`.